packages/storage/src/tasksnapshot.ts

```diff
@@ -16,16 +16,19 @@
  */
 import { TaskState } from './implementation/taskenums';
 import { Metadata } from './metadata';
-import { Reference } from './reference';
+import { StorageReference } from './reference';
 import { UploadTask } from './task';
 
-export class UploadTaskSnapshot {
-  constructor(
-    readonly bytesTransferred: number,
-    readonly totalBytes: number,
-    readonly state: TaskState,
-    readonly metadata: Metadata,
-    readonly task: UploadTask,
-    readonly ref: Reference
-  ) {}
+export interface UploadTaskSnapshot {
+  readonly metadata: Metadata;
+  readonly ref: StorageReference;
+}
+
+export interface UploadTaskResumableSnapshot {
+  readonly bytesTransferred: number;
+  readonly totalBytes: number;
+  readonly state: TaskState;
+  readonly metadata: Metadata;
+  readonly task: UploadTask;
+  readonly ref: StorageReference;
```

Comment:
I wonder if these are the best names. Have you consider `UploadTaskSnapshot` and `UploadResult` or similar? The reason I ask is:
- `UploadTaskSnapshot` isn't really a snapshot anymore, it's the end state. Plus, the new upload format doesn't actually involve UploadTasks anymore,
- `UploadTaskResumableSnapshot` doesn't offer any APIs or method that let you pause or resume an Upload. Instead, these methods live on UploadTask and don't depend on this snapshot.

Reply:
+1

Reply:
Sounds like a good plan.